tests/phpunit/AutoReview/Makefile/MakefileTest.php

```diff
@@ -278,84 +254,149 @@ static function (array $targetSet) use ($target, $dashCount): bool {
-            Parser::parse(file_get_contents(self::MAKEFILE_PATH)),
-            static function (array $targetSet): bool {
-                [$target, $dependencies] = $targetSet;
-
-                return strpos($target, 'test') === 0
-                    && substr($target, -7) === '-docker'
-                    && ($dependencies === []
-                        || strpos($dependencies[0], '## ') !== 0
-                    )
-                ;
-            }
+        $testRules = self::getTestRules(true);
+
+        $testPrerequisites = array_shift($testRules)[1];
+
+        $rootTestTargets = self::getRootTestTargets($testRules, 2);
+        array_unshift($rootTestTargets, 'autoreview');
+
+        $this->assertEqualsCanonicalizing($rootTestTargets, $testPrerequisites);
+    }
+
+    private static function executeMakeCommand(string $commandName): string
+    {
+        $timeout = self::getTimeout();
+
+        return self::executeCommand(
+            sprintf(
+                '%s make %s help --silent --file %s 2>&1',
+                $timeout,
+                $commandName,
+                self::MAKEFILE_PATH,
+            ),
         );
+    }
+
+    // TODO: remove this as we remove support for PHP 7.4 and Safe v1
+    private static function executeCommand(string $command): string
+    {
+        if (function_exists('Safe\shell_exec')) {
+            return shell_exec($command);
+        }
+
+        error_clear_last();
+
+        $safeResult = unsafe_shell_exec($command);
+
+        if ($safeResult === null || $safeResult === false) {
+            throw ExecException::createFromPhpError();
+        }
+
+        return $safeResult;
+    }
 
-        $testDependencies = array_shift($testTargets)[1];
+    private static function getTimeout(): string
+    {
+        try {
+            self::executeCommand('command -v timeout');
 
-        $rootTestTargets = array_column(
+            return 'timeout 2s';
+        } catch (ExecException $execException) {
+            return '';
+        }
+    }
+
+    /**
+     * @return list<array{string, list<string>}>
+     */
+    private static function getTestRules(bool $dockerTargets): array
+    {
+        $filterDockerTarget = $dockerTargets
+            ? static fn (string $target) => substr($target, -7) === '-docker'
+            : static fn (string $target) => substr($target, -7) !== '-docker';
+
+        return array_values(
             array_filter(
-                $testTargets,
-                static function (array $targetSet): bool {
-                    return strpos($targetSet[0], 'test-') === 0
-                        && substr_count($targetSet[0], '-') === 2;
+                Parser::parse(file_get_contents(self::MAKEFILE_PATH)),
+                static function (array $rule) use ($filterDockerTarget): bool {
+                    [$target, $prerequisites] = $rule;
+
+                    $isCommentRule = count($prerequisites) !== 0 && str_starts_with($prerequisites[0], '## ');
```

Comment:
note that I replaced a few occurrences of `strpos()` to `str_starts_with`. I thought we had the polyfills setup but apparently this is not the case so this should be reverted

Reply:
Actually not a problem: the auto-review tests are run on PHP 8.0